dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPIImpl.java

```diff
@@ -83,37 +54,39 @@ public FieldAPIImpl(final PermissionAPI perAPI,
                         final ContentletAPI conAPI,
                         final UserAPI userAPI,
                         final DeleteFieldJobHelper deleteFieldJobHelper) {
-      this.perAPI = perAPI;
-      this.conAPI = conAPI;
+      this.permissionAPI = perAPI;
+      this.contentletAPI = conAPI;
       this.userAPI = userAPI;
   }
 
+  @WrapInTransaction
   @Override
-	public Field save(Field field, User user) throws DotDataException, DotSecurityException {
+  public Field save(final Field field, final User user) throws DotDataException, DotSecurityException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)